ui/server/src/test/scala/pl/touk/nussknacker/ui/api/ManagementResourcesConcurentSpec.scala

```diff
@@ -3,61 +3,79 @@ package pl.touk.nussknacker.ui.api
+  private val processId = SampleProcess.process.id
 
-    saveProcessAndAssertSuccess(SampleProcess.process.id, SampleProcess.process)
+  test("not allow concurrent deployment of same process") {
 
-    def invokeDeploy =
-      deployProcess(SampleProcess.process.id) ~> runRoute
+    saveProcessAndAssertSuccess(processId, SampleProcess.process)
 
-    val firstRun = invokeDeploy
-    invokeDeploy ~> check {
-      status shouldBe StatusCodes.Conflict
-    }
-    firstRun ~> check {
-      status shouldBe StatusCodes.OK
+    withWaitForDeployFinish(processId) {
+      eventually {
+        deployProcess(processId) ~> runRoute ~> check {
+          status shouldBe StatusCodes.Conflict
+        }
+      }
     }
-    invokeDeploy ~> check {
+    deployProcess(processId) ~> runRoute ~> check {
       status shouldBe StatusCodes.OK
     }
   }
 
-  ignore("not allow concurrent deployment and cancel of same process") {
+  test("not allow concurrent deployment and cancel of same process") {
     saveProcessAndAssertSuccess(SampleProcess.process.id, SampleProcess.process)
 
-      val firstRun = deployProcess(SampleProcess.process.id) ~> runRoute
-      cancelProcess(SampleProcess.process.id) ~> check {
-        status shouldBe StatusCodes.Conflict
-      }
-      firstRun ~> check {
-        status shouldBe StatusCodes.OK
+    withWaitForDeployFinish(processId) {
+      eventually {
+        cancelProcess(processId) ~> check {
+          status shouldBe StatusCodes.Conflict
+        }
       }
-      cancelProcess(SampleProcess.process.id) ~> check {
-        status shouldBe StatusCodes.OK
-      }
-
+    }
+    cancelProcess(SampleProcess.process.id) ~> check {
+      status shouldBe StatusCodes.OK
+    }
   }
 
-  ignore("not allow concurrent deployment of different processes") {
-    saveProcessAndAssertSuccess(SampleProcess.process.id, SampleProcess.process)
+  test("not allow concurrent deployment of different processes") {
+    saveProcessAndAssertSuccess(processId, SampleProcess.process)
 
     val secondId = SampleProcess.process.id + "-2"
     saveProcessAndAssertSuccess(secondId, SampleProcess.process)
 
-      val firstRun = deployProcess(SampleProcess.process.id) ~> runRoute
-      deployProcess(secondId) ~> check {
-        status shouldBe StatusCodes.Conflict
+    withWaitForDeployFinish(processId) {
+      eventually {
+        deployProcess(secondId) ~> check {
+          status shouldBe StatusCodes.Conflict
+        }
       }
-      runRoute ~> check {
-        status shouldBe StatusCodes.OK
+    }
+
+    deployProcess(secondId) ~> runRoute ~> check {
+      status shouldBe StatusCodes.OK
+    }
+
+  }
+
+  private def withWaitForDeployFinish(id: String)(action: => Unit): Unit = {
+    val firstRun = deploymentManager.withWaitForDeployFinish {
+      val firstRun = deployProcess(id) ~> runRoute
+      eventually {
+        deploymentManager.deploys.get() should not be 'empty
```

Comment:
We should take a look into this list to make sure that given `id` was deployed?

Reply:
yes, I added comment & additional check